common/ui/file.cpp

```diff
@@ -41,7 +41,7 @@ static PHYSFSX_counted_list file_getdirlist(const char *dir)
 {
 	ntstring<PATH_MAX - 1> path;
 	auto dlen = path.copy_if(dir);
-	if (!dlen || !path.copy_if(dlen, "/"))
+	if ((!dlen && strlen(dir) > 0) || !path.copy_if(dlen, "/"))
```

Comment:
Please do not use `strlen` just to test if a string is empty.  Instead, check whether the first byte is zero.